src/Microsoft.Health.Fhir.Shared.Api.UnitTests/Features/Exceptions/ExceptionNotificationMiddlewareTests.cs

```diff
@@ -0,0 +1,83 @@
+    {
+        private readonly DefaultHttpContext _context;
+        private readonly IFhirRequestContextAccessor _fhirRequestContextAccessor;
+        private readonly IMediator _mediator = Substitute.For<IMediator>();
+
+        public ExceptionNotificationMiddlewareTests()
+        {
+            _fhirRequestContextAccessor = Substitute.For<IFhirRequestContextAccessor>();
+            _context = new DefaultHttpContext();
+        }
+
+        [Fact]
+        public async Task GivenAnHttpContextWithException_WhenExecutingExceptionNotificationMiddleware_MediatRShouldEmitNotification()
+        {
+            var exceptionMessage = "Test exception";
+            var exceptionNotificationMiddleware = CreateExceptionNotificationMiddleware(innerHttpContext => throw new Exception(exceptionMessage));
+
+            try
+            {
+                await exceptionNotificationMiddleware.Invoke(_context);
+            }
+            catch (Exception e)
+            {
+                await _mediator.ReceivedWithAnyArgs(1).Publish(Arg.Any<ExceptionNotification>(), Arg.Any<CancellationToken>());
+                Assert.Equal(exceptionMessage, e.Message);
+            }
+        }
+
+        [Fact]
+        public async Task GivenAnHttpContextWithNoException_WhenExecutingExceptionNotificationMiddleware_MediatRShouldNotEmitNotification()
+        {
+            var exceptionNotificationMiddleware = CreateExceptionNotificationMiddleware(innerHttpContext => Task.CompletedTask);
+
+            await exceptionNotificationMiddleware.Invoke(_context);
+
+            await _mediator.DidNotReceiveWithAnyArgs().Publish(Arg.Any<object>(), Arg.Any<CancellationToken>());
+        }
+
+        [Fact]
+        public async Task WhenMediatRFails_OriginalExceptionStillThrown()
+        {
+            var exceptionMessage = "Test exception";
+            var mediatorExceptionMessage = "Mediator Failure";
+
+            var exceptionNotificationMiddleware = CreateExceptionNotificationMiddleware(innerHttpContext => throw new Exception(exceptionMessage));
+            _mediator.WhenForAnyArgs(async x => await x.Publish(Arg.Any<ExceptionNotificationMiddleware>(), Arg.Any<CancellationToken>())).Throw(new System.Exception(mediatorExceptionMessage));
+
+            try
+            {
+                await exceptionNotificationMiddleware.Invoke(_context);
+            }
+            catch (Exception e)
+            {
+                await _mediator.DidNotReceiveWithAnyArgs().Publish(Arg.Any<object>(), Arg.Any<CancellationToken>());
+                Assert.Equal(exceptionMessage, e.Message);
+            }
+        }
+
+        private ExceptionNotificationMiddleware CreateExceptionNotificationMiddleware(RequestDelegate nextDelegate)
+        {
+            return Substitute.ForPartsOf<ExceptionNotificationMiddleware>(nextDelegate, NullLogger<ExceptionNotificationMiddleware>.Instance, _fhirRequestContextAccessor, _mediator);
```

Comment:
Why does this need to be a substitute? It doesn't look like the Substitute methods are used on the test class.

Reply:
It doesn't, I was thinking about doing some other tests, but ended up not doing them. I removed the substitution.